Tella/Data/Networking/APICall/URLRequest.swift

```diff
@@ -57,8 +57,8 @@ extension Publisher where Output == URLSession.DataTaskPublisher.Output {
             }
             return ($0.0, ($0.1 as? HTTPURLResponse)?.allHeaderFields)
         }
-        .mapError{  error in
-            return error as! APIError
+        .mapError{ error in
+            APIError.error(error.localizedDescription)
```

Comment:
What do you think if you change to 
```
        .mapError{ error in
            let error = (error as NSError)
            return APIError.httpCode(error.code)
        }

```


and then add  badServer = -1003

```
enum HTTPErrorCodes: Int {
    case badRequest = 400
    case unauthorized = 401
    case forbidden = 403
    case notFound = 404
    case methodNotAllowed = 405
    case requestTimeout = 408
    case need2FA = 409
    case internalServerError = 500
    case badServer = -1003
    case unknown
}

```


then update the customErrorMessage


```
    private func customErrorMessage(errorCode : Int) -> String {
        switch HTTPErrorCodes(rawValue: errorCode) {
        case .unauthorized:
            return "Invalid username or password"
        case .forbidden:
            return "Account locked due to too many unsuccessful attempts."
        case .badServer:
            return "Error: The server URL is incorrect"
        default:
            return "Custom Error"
        }
    }

```

then 

    fileprivate func handleCompletionForCheckURL(_ completion: Subscribers.Completion<APIError>) {
        self.isLoading = false
        switch completion {
        case .finished:
            debugLog("Finished")
            // TODO: handle this error
        case .failure(let error):
            switch error {
            case .noInternetConnection:
                Toast.displayToast(message: error.errorDescription ?? error.localizedDescription)
            default:
                debugLog(error)
                urlErrorMessage = error.errorDescription ?? error.localizedDescription
                shouldShowURLError = true
            }
        }
    }

Reply:
The server is sending us `404` status code for instead of making a new enum type for `HTTPErrorCodes` i have used the `notFound ` case and made change in that conditions  like
```swift
private func customErrorMessage(errorCode : Int) -> String {
        switch HTTPErrorCodes(rawValue: errorCode) {
        case .unauthorized:
            return "Invalid username or password"
        case .forbidden:
            return "Account locked due to too many unsuccessful attempts."
        case .notFound:
            return "Error: The server URL is incorrect"
        default:
            return "Custom Error"
        }
    }
```

and the 
```swift
.mapError{ error in
            let error = (error as NSError)
            return APIError.httpCode(error.code)
        }
``` 
is changed to 
```swift 
.mapError{ error in
            return error as! APIError
        }
```
as the `tryMap` operator above is throwing the `APIError` only so `APIError.httpCode(error.code)` is not needed IMO